src/main/kotlin/com/cognifide/gradle/aem/instance/content/Node.kt

```diff
@@ -0,0 +1,90 @@
+package com.cognifide.gradle.aem.instance.content
+
+import com.cognifide.gradle.aem.common.http.RequestException
+import com.cognifide.gradle.aem.instance.InstanceSync
+import com.jayway.jsonpath.DocumentContext
+import net.minidev.json.JSONArray
+import org.apache.jackrabbit.vault.util.JcrConstants
+
+class Node private constructor(val sync: InstanceSync, val path: String, val props: Map<String, Any> = mapOf()) {
+
+    lateinit var document: DocumentContext
+
+    val name: String
+        get() = path.substringAfterLast("/")
+    val jcrTitle: String
+        get() = stringValue(JcrConstants.JCR_TITLE)
+    val jcrCreated
+        get() = value(JcrConstants.JCR_CREATED)
+    val jcrCreatedBy: String
+        get() = stringValue(JcrConstants.JCR_CREATED_BY)
+    val jcrDescription: String
+        get() = stringValue(JcrConstants.JCR_DESCRIPTION)
+    val jcrPrimaryType: String
+        get() = stringValue(JcrConstants.JCR_PRIMARYTYPE)
+    val children: Iterator<Node>
+        get() = sync.get("$path.harray.1.json") { asJson(it) }
+                .read<JSONArray>("__children__")
+                .map { child -> child as Map<*, *> }
+                .map { props -> load(sync, path + "/" + props["__name__"] as String) }
+                .iterator()
+
+    fun value(propName: String): Any = document.read(propName) as Any
+    fun stringValue(propName: String): String = document.read(propName) as String
+    fun json(): String = document.jsonString()
+
+    companion object {
+        fun load(sync: InstanceSync, path: String): Node = Node(sync, path).apply {
```

Comment:
after that we decided to abandon `ObjectMapper` approach and hiding Node constructor, maybe eliminate wrappers in `InstanceRepository` / copy these methods there